src/manifest.rs

```
@@ -864,15 +864,25 @@ fn default_manifest(input: &Input, input_id: &OsString) -> MainResult<toml::valu
 }
 
 /**
-Given two Cargo manifests, merges the second *into* the first.
+Given two Cargo manifests, merges the second (user's) *into* the first (default).
 
-Note that the "merge" in this case is relatively simple: only *top-level* tables are actually merged; everything else is just outright replaced.
+Note that the "merge" operates on *top-level* tables and only adds at that level.
+
+Mainly we only care about the \[package\] and \[bin\] sections because they are in the default manifest.
+The original comments stated "everything else is just outright replaced" this was not exactly correct.
+The "info_t.extend(from_t);" call will add new map entries and overwrite map entries with the same name.
+This gives us a sub_element merger. Everything int the default \[package\] is up for game.
+
+We preserve bin however since changing this would likely break compilation.
 */
 fn merge_manifest(
     mut into_t: toml::value::Table,
     from_t: toml::value::Table,
 ) -> MainResult<toml::value::Table> {
     for (k, v) in from_t {
+        if k == "bin" {
+            continue;
+        } // Skip [bin] to prevent a user from shooting themselves in the foot
```

I think I'd rather error than silently ignore

---

Sure, we can error here.